aquacomputer_d5next.c

```diff
@@ -1710,6 +1733,13 @@ static int aqc_raw_event(struct hid_device *hdev, struct hid_report *report, u8
 	/* Special-case sensor readings */
 	switch (priv->kind) {
 	case aquaero:
+		/* Read hardware version (for v5: 5600, for v6: 6000, usually) */
+		priv->aquaero_hw_version = get_unaligned_be16(data + AQUAERO_HARDWARE_VERSION);
+		if (priv->aquaero_hw_version / 1000 == 5)	/* Get the first digit */
+			priv->aquaero_hw_kind = aquaero5;
+		else
+			priv->aquaero_hw_kind = aquaero6;
```

Comment:
Do we want Aquaero 6 to be the default in case another hardware revision appears?
I would prefer to check specifically check for the known values.

Reply:
Makes sense, will revise.